sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RuleDescriptionSection.java

```diff
@@ -32,8 +32,9 @@ public interface RuleDescriptionSection {
    */
   class RuleDescriptionSectionKeys {
     public static final String INTRODUCTION_SECTION_KEY = "introduction";
-    public static final String WHY_IS_THIS_AN_ISSUE_SECTION_KEY = "why_is_it_an_issue";
-    public static final String HOW_TO_FIX_IT_SECTION_KEY = "how_to_fix_it";
+    public static final String WHY_SECTION_KEY = "why";
+    public static final String ASSESS_SECTION_KEY = "assess";
+    public static final String HOW_SECTION_KEY = "how";
```

Comment:
Just to be sure I understood the change fully, by making these sections generic, the `why` section would represent for vulnerability the `Why is this an issue ?` tab, and the `What's the risk ?` tab for hotspot ?
And the new `assess` section represents the hotspot `Are you at risk ?` tab?

Reply:
The `assess` key would be for the current hotspot description tab "Assess the risk".

The `why` would indeed map to both "why is this an issue" and "what's the risk", depending on the issue type; But I realise now it's a bit confusing, maybe we can come up with a better identifier...

Reply:
Ok, thanks for the clarification !
If we want to use the same identifier for both "why is this an issue" and "what's the risk" tabs depending on issue type, I think it'd indeed make sense to use an identifier that is meaningful to both.
Although I must admit I don't have any idea for a better name at the moment

Reply:
What about `the_problem`?

Reply:
I have just pushed a new proposal:

```
    public static final String INTRODUCTION_SECTION_KEY = "introduction";
    public static final String THE_PROBLEM_SECTION_KEY = "the_problem";
    public static final String ASSESS_THE_PROBLEM_SECTION_KEY = "assess_the_problem";
    public static final String HOW_TO_FIX_SECTION_KEY = "how_to_fix";
    public static final String RESOURCES_SECTION_KEY = "resources";
```

Reply:
I think it would be more inline with our current naming to use `issue` instead of `problem`, no?
How about `issue_root_cause` or `cause_of_issue` ?
I'm not very gifted with naming to be honest^^
As long as it is clear, to be honest, it's fine on my side

Reply:
Actually I'm precisely trying to avoid the word `issue` as it is confusing in the case of hotspots :/

Reply:
@christophe-zurn-sonarsource suggested to use `root_cause`; we also believe it is a nice key so I'll modify the PR accordingly and merge